src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs

```diff
@@ -670,6 +670,124 @@ public void IdFormatTests()
+                Assert.True(activity.Id[0] == '|');
+                activity.Stop();
+
+                // can set and override format before activity starts                
+                activity = new Activity("activity5");
+                Assert.Equal(ActivityIdFormat.Unknown, activity.IdFormat);
+                activity.SetIdFormat(ActivityIdFormat.Hierarchical);
+                Assert.Equal(ActivityIdFormat.Hierarchical, activity.IdFormat);
+                activity.SetIdFormat(ActivityIdFormat.W3C);
+                Assert.Equal(ActivityIdFormat.W3C, activity.IdFormat);
+
+                // SetIdFormat overrides forced W3C DefaultIdFormat
+                Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+                Activity.ForceDefaultIdFormat = true;
+                activity = new Activity("activity7");
+                activity.SetIdFormat(ActivityIdFormat.Hierarchical);
+                activity.Start();
+                Assert.Equal(ActivityIdFormat.Hierarchical, activity.IdFormat);
+                activity.Stop();
+                Activity.ForceDefaultIdFormat = false;
+                Activity.DefaultIdFormat = ActivityIdFormat.Hierarchical;
+
+                // SetIdFormat overrides forced Hierarchical DefaultIdFormat 
+                Activity.DefaultIdFormat = ActivityIdFormat.Hierarchical;
+                Activity.ForceDefaultIdFormat = true;
+                activity = new Activity("activity8");
+                activity.SetIdFormat(ActivityIdFormat.W3C);
+                activity.Start();
+                Assert.Equal(ActivityIdFormat.W3C, activity.IdFormat);
+                activity.Stop();
+                Activity.ForceDefaultIdFormat = false;
+                Activity.DefaultIdFormat = ActivityIdFormat.Hierarchical;
+
+                // SetIdFormat overrides Hierarchical parent format 
+                Activity parent = new Activity("parent")
+                    .SetIdFormat(ActivityIdFormat.Hierarchical)
+                    .Start();
+
+                Activity child = new Activity("child")
+                    .SetIdFormat(ActivityIdFormat.W3C)
+                    .Start();
+
+                Assert.Equal(ActivityIdFormat.W3C, child.IdFormat);
+                child.Stop();
+                parent.Stop();
+
+                // SetIdFormat overrides W3C Parent format 
+                parent = new Activity("parent")
+                    .SetIdFormat(ActivityIdFormat.W3C)
+                    .Start();
+
+                child = new Activity("child")
+                    .SetIdFormat(ActivityIdFormat.Hierarchical)
+                    .Start();
+
+                Assert.Equal(ActivityIdFormat.Hierarchical, child.IdFormat);
+                child.Stop();
+                parent.Stop();
+            }
+            finally
+            {
+                // Set global settings back to the default, just to put the state back. 
+                Activity.ForceDefaultIdFormat = false;
```

Comment:
These fields are a bit scary to set in a unit test, don't these run in parallel?